api/analyzedphenotypes.cv.api.inc

```diff
@@ -0,0 +1,547 @@
+        );
+      }
+    }
+  }
+
+  return $cvterm;
+}
+
+/**
+ * FUNCTION:
+ * Insert an entry into chado.cvterm table.
+ * Dependencies: ap_get_variablenames() - systemvars.api.
+ *
+ * @param $identifiers
+ *   An array with the following keys:
+ *   - name : controlled vocabylary name/trait name/column header.
+ *   - definition : human-readable definition of a term.
+ *   - genus : genus.
+ *
+ *   An array to be used as additional options to be performed to
+ *   returned result. Keys:
+ *   - return_inserted_id : option to return cvterm term id number inserted.
+ *
+ * @return
+ *   An array, term properties.
+ */
+function ap_insert_cvterm($identifiers, $options = array('return_inserted_id' => TRUE)) {
+  $sysvar = ap_get_variablenames(
+    array('variablename' => $identifiers['genus']),
+    array('set' => 'cvdbon', 'suffix' => 'allsuffix')
+  );
+
+  // Database.
+  $sysvar_genus_db = variable_get($sysvar['db']);
+  //
+  // TODO: Replace when database API is ready.
+  // Requires the db name.
+  $dbprop = analyzedphenotypes_dbprop('match_db', array(
+    'db_id' => $sysvar_genus_db,
+  ));
+  $db_name = $dbprop['name'];
+
+  // Controlled Vocabulary.
+  $sysvar_genus_cv = variable_get($sysvar['cv']);
+  $cvprop = analyzedphenotypes_cvprop('match_cvid', array(
+    'cv_id' => $sysvar_genus_cv,
+  ));
+  $cv_name = $cvprop['name'];
+
+  //Insert cvterm.
+  $cvterm = tripal_insert_cvterm(array(
+    'id'  => trim($db_name) . ':' . $identifiers['name'],
+    'name'  => $identifiers['name'],
+    'cv_name' => $cv_name,
+    'definition' => $identifiers['definition']
+  ));
+
+  // Return inserted id when implied in options.
+  if ($cvterm && $options['return_inserted_id']) {
+    return $cvterm->cvterm_id;
+  }
+}
+
+/**
+ * FUNCTION:
+ * Upload/save image/photo of a cvterm.
+ * Dependencies:
+ *
+ * @param $identifiers
+ *   An array with the following keys:
+ *   - cvterm_id : cvterm id number of term/trait.
+ *   - photo : File field API value name ap_photo_1 and ap_photo_2.
+ *
+ * @param $options
+ *   An array to be used as additional options to be performed to
+ *   returned result. Keys:
+ *   - scaleto : scale to or reduce image size to this value. Default to 400X400 px.
+ */
+function ap_upload_cvtermphoto($identifiers, $options = array('scaleto' => 400)) {
+  foreach($identifiers['photo'] as $i => $photo) {
+    // Save upload file.
+    $tmp_file = file_save_upload($photo, array(), FALSE, FILE_EXISTS_REPLACE);
+
+    // Access file saved and modify dimension plus filename plus filename.
+    if ($tmp_file) {
+      $file_obj = file_load($tmp_file->fid);
+
+      $source_file = drupal_realpath($file_obj->uri);
+      $info = image_get_info($source_file);
+
+      // Scale but maintain aspect ratio.
+      $scale = analyzedphenotypes_scale_photo(array(
+        'height' => $info['height'],
+        'width' => $info['width'],
+        'scale_to' => $options['scaleto'],
+       ));
+
+      $photo_dir = 'public://ap-photo';
```

Comment:
You should be really particular about where you save files ;-). A better path would be `public://tripal/analyzedphenotypes/trait_pictures`. You would need to make sure the directory structure ahead of time though.